### PR TITLE
Add Rk3Pareschi IMEX time stepper

### PR DESCRIPTION
**Pull request by wthrowe**
Note to @yoonso0-0: I've renamed the class since the last version I gave you.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Time/TimeSteppers/Rk3Pareschi.cpp [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+       {0.0},
+       {0.0, 0.0, 1.0},
+       {0.0, 0.0, 1.0 / 4.0, 1.0 / 4.0}},
+      // Result coefficients
+      {0.0, 0.0, 1.0 / 6.0, 1.0 / 6.0, 2.0 / 3.0},
+      // Coefficients for the embedded method for generating an error measure.
+      //
+      // Not given in the reference.  Any set of coefficients of the form
+      // (0, 0, x, x, 1 - 2 x) works.
+      {0.0, 0.0, 0.0, 0.0, 1.0},
```
Comment (teukolsky) [inline src/Time/TimeSteppers/Rk3Pareschi.cpp]: It's probably better to use a non-zero value of x so the error estimate samples the y-values across the interval. x=1/4 looks good. Using only the last value is probably not robust. This is the reason, by the way, that Dormand-Prince is 853 and not, say 87 or 86. These 'simpler' error estimates turned out not to sample the interval very well.

File: src/Time/TimeSteppers/Rk3Pareschi.hpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+#include "Options/String.hpp"
+#include "Time/TimeSteppers/ImexRungeKutta.hpp"
+#include "Utilities/Serialization/CharmPupable.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace TimeSteppers {
+/*!
+ * \ingroup TimeSteppersGroup
+ * \brief A fourth-order Runge-Kutta method with IMEX support.
```
Comment (nilsdeppe) [inline src/Time/TimeSteppers/Rk3Pareschi.hpp]: fourth -> third?

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
LGTM, one small question that might be a typo. Please squash and rebase immediately :)

File: src/Time/TimeSteppers/Rk3Pareschi.hpp (hunk 2 above)
Comment (wthrowe) [reply src/Time/TimeSteppers/Rk3Pareschi.hpp]: Yep.  Fixed.

**Review by nilsdeppe — APPROVED** [review]

**Comment by nilsdeppe** [comment]
CI failure is unrelated.

**Pull request merged**